### PR TITLE
fix(manager/sbt): handle new line after equalities in the file

### LINKED ISSUE #14846: Dependencies are skipped depending on whitespace
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.9.3

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Minimal reproduction repo:
https://github.com/nilius/renovate_issue

With this scala build.sbt file, dependencies ARE NOT updated:

```
name := "service"
scalaVersion := "2.13.8"

lazy val compileDependencies = 
  Seq(
    "com.typesafe.scala-logging" %% "scala-logging" % "3.9.4",
    "ch.qos.logback" % "logback-classic" % "1.2.10"
  )

libraryDependencies ++= compileDependencies
```

With this scala build.sbt file, dependencies ARE updated:
```
name := "service"
scalaVersion := "2.13.8"

lazy val compileDependencies = Seq(
    "com.typesafe.scala-logging" %% "scala-logging" % "3.9.4",
    "ch.qos.logback" % "logback-classic" % "1.2.10"
  )

libraryDependencies ++= compileDependencies
```

The only difference is in the newline before the "Seq"!

You can see in the debug logs, that the dependencies "scala-logging" and "logback-classic" are missing completely in the first case, the whole block seems to be skipped silently.

Dependency updates should not depend on the whitespace of the build.sbt, this is completely unexpected.

### Relevant debug logs

<details><summary>Logs</summary>

```
 ==================== LOG for NOK case ====================

DEBUG: migrated config (repository=nilius/renovate_issue)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "labels": ["renovate"],
         "timezone": "Europe/Berlin",
         "schedule": ["after 9am and before 5pm on Mon to Fri"],
         "updateNotScheduled": true,
         "dependencyDashboard": false,
         "prHourlyLimit": 0,
         "prConcurrentLimit": 1,
         "prCreation": "not-pending",
         "recreateClosed": false,
         "rebaseWhen": "behind-base-branch",
         "separateMajorMinor": false,
         "semanticCommits": "disabled",
         "stabilityDays": 0,
         "rangeStrategy": "replace",
         "enabledManagers": ["sbt"],
         "packageRules": [
           {
             "matchDatasources": ["sbt-package"],
             "registryUrls": ["https://repo.maven.apache.org/maven2"]
           },
           {
             "matchDatasources": ["sbt-plugin"],
             "registryUrls": ["https://repo.maven.apache.org/maven2"]
           },
           {
             "groupName": "all versions",
             "matchUpdateTypes": ["major", "minor", "patch", "pin", "digest"],
             "automerge": true
           },
           {
             "matchManagers": ["sbt"],
             "matchPackagePrefixes": ["io.confluent"],
             "registryUrls": ["https://packages.confluent.io/maven/"]
           }
         ]
       }
DEBUG: Found repo ignorePaths (repository=nilius/renovate_issue)
       "ignorePaths": ["**/node_modules/**", "**/bower_components/**"]
DEBUG: No vulnerability alerts found (repository=nilius/renovate_issue)
DEBUG: No vulnerability alerts found (repository=nilius/renovate_issue)
DEBUG: No baseBranches (repository=nilius/renovate_issue)
DEBUG: extract() (repository=nilius/renovate_issue)
DEBUG: Setting current branch to main (repository=nilius/renovate_issue)
DEBUG: latest commit (repository=nilius/renovate_issue)
       "branchName": "main",
       "latestCommitDate": "2022-03-29T12:51:03+02:00"
DEBUG: Applying enabledManagers filtering (repository=nilius/renovate_issue)
DEBUG: Using file match: \.sbt$ for manager sbt (repository=nilius/renovate_issue)
DEBUG: Using file match: project/[^/]*.scala$ for manager sbt (repository=nilius/renovate_issue)
DEBUG: Matched 1 file(s) for manager sbt: build.sbt (repository=nilius/renovate_issue)
DEBUG: Found sbt package files (repository=nilius/renovate_issue)
DEBUG: Found 1 package file(s) (repository=nilius/renovate_issue)
 INFO: Dependency extraction complete (repository=nilius/renovate_issue)
       "baseBranch": "main",
       "stats": {
         "managers": {"sbt": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
DEBUG: Looking up org.scala-lang:scala-library in repository https://repo.maven.apache.org/maven2/ (repository=nilius/renovate_issue)
DEBUG: Found 194 new releases for org.scala-lang:scala-library in repository https://repo.maven.apache.org/maven2/ (repository=nilius/renovate_issue)
DEBUG: Package releases lookups complete (repository=nilius/renovate_issue)
       "baseBranch": "main"
DEBUG: branchifyUpgrades (repository=nilius/renovate_issue)
DEBUG: 0 flattened updates found:  (repository=nilius/renovate_issue)
DEBUG: Returning 0 branch(es) (repository=nilius/renovate_issue)
DEBUG: config.repoIsOnboarded=true (repository=nilius/renovate_issue)
DEBUG: packageFiles with updates (repository=nilius/renovate_issue)
       "config": {
         "sbt": [
           {
             "packageFile": "build.sbt",
             "deps": [
               {
                 "registryUrls": ["https://repo.maven.apache.org/maven2"],
                 "datasource": "maven",
                 "depName": "scala",
                 "packageName": "org.scala-lang:scala-library",
                 "currentValue": "2.13.8",
                 "separateMinorPatch": true,
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "ivy",
                 "sourceUrl": "https://github.com/scala/scala",
                 "homepage": "https://www.scala-lang.org/",
                 "currentVersion": "2.13.8",
                 "fixedVersion": "2.13.8"
               }
             ]
           }
         ]
       }
DEBUG: processRepo() (repository=nilius/renovate_issue)
DEBUG: Processing 0 branches:  (repository=nilius/renovate_issue)

 ==================== LOG for OK case (2nd build.sbt) ====================

DEBUG: migrated config (repository=nilius/renovate_issue)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "labels": ["renovate"],
         "timezone": "Europe/Berlin",
         "schedule": ["after 9am and before 5pm on Mon to Fri"],
         "updateNotScheduled": true,
         "dependencyDashboard": false,
         "prHourlyLimit": 0,
         "prConcurrentLimit": 1,
         "prCreation": "not-pending",
         "recreateClosed": false,
         "rebaseWhen": "behind-base-branch",
         "separateMajorMinor": false,
         "semanticCommits": "disabled",
         "stabilityDays": 0,
         "rangeStrategy": "replace",
         "enabledManagers": ["sbt"],
         "packageRules": [
           {
             "matchDatasources": ["sbt-package"],
             "registryUrls": ["https://repo.maven.apache.org/maven2"]
           },
           {
             "matchDatasources": ["sbt-plugin"],
             "registryUrls": ["https://repo.maven.apache.org/maven2"]
           },
           {
             "groupName": "all versions",
             "matchUpdateTypes": ["major", "minor", "patch", "pin", "digest"],
             "automerge": true
           },
           {
             "matchManagers": ["sbt"],
             "matchPackagePrefixes": ["io.confluent"],
             "registryUrls": ["https://packages.confluent.io/maven/"]
           }
         ]
       }
DEBUG: Found repo ignorePaths (repository=nilius/renovate_issue)
       "ignorePaths": ["**/node_modules/**", "**/bower_components/**"]
DEBUG: No vulnerability alerts found (repository=nilius/renovate_issue)
DEBUG: No vulnerability alerts found (repository=nilius/renovate_issue)
DEBUG: No baseBranches (repository=nilius/renovate_issue)
DEBUG: extract() (repository=nilius/renovate_issue)
DEBUG: Setting current branch to main (repository=nilius/renovate_issue)
DEBUG: latest commit (repository=nilius/renovate_issue)
       "branchName": "main",
       "latestCommitDate": "2022-03-29T12:50:10+02:00"
DEBUG: Applying enabledManagers filtering (repository=nilius/renovate_issue)
DEBUG: Using file match: \.sbt$ for manager sbt (repository=nilius/renovate_issue)
DEBUG: Using file match: project/[^/]*.scala$ for manager sbt (repository=nilius/renovate_issue)
DEBUG: Matched 1 file(s) for manager sbt: build.sbt (repository=nilius/renovate_issue)
DEBUG: Found sbt package files (repository=nilius/renovate_issue)
DEBUG: Found 1 package file(s) (repository=nilius/renovate_issue)
 INFO: Dependency extraction complete (repository=nilius/renovate_issue)
       "baseBranch": "main",
       "stats": {
         "managers": {"sbt": {"fileCount": 1, "depCount": 3}},
         "total": {"fileCount": 1, "depCount": 3}
       }
DEBUG: Looking up org.scala-lang:scala-library in repository https://repo.maven.apache.org/maven2/ (repository=nilius/renovate_issue)
DEBUG: Found 194 new releases for org.scala-lang:scala-library in repository https://repo.maven.apache.org/maven2/ (repository=nilius/renovate_issue)
DEBUG: Package releases lookups complete (repository=nilius/renovate_issue)
       "baseBranch": "main"
DEBUG: branchifyUpgrades (repository=nilius/renovate_issue)
DEBUG: Using group branchName template (repository=nilius/renovate_issue)
DEBUG: Dependency ch.qos.logback:logback-classic is part of group all versions (repository=nilius/renovate_issue)
DEBUG: 1 flattened updates found: ch.qos.logback:logback-classic (repository=nilius/renovate_issue)
DEBUG: Returning 1 branch(es) (repository=nilius/renovate_issue)
DEBUG: config.repoIsOnboarded=true (repository=nilius/renovate_issue)
DEBUG: packageFiles with updates (repository=nilius/renovate_issue)
       "config": {
         "sbt": [
           {
             "packageFile": "build.sbt",
             "deps": [
               {
                 "registryUrls": ["https://repo.maven.apache.org/maven2"],
                 "datasource": "maven",
                 "depName": "scala",
                 "packageName": "org.scala-lang:scala-library",
                 "currentValue": "2.13.8",
                 "separateMinorPatch": true,
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [],
                 "versioning": "ivy",
                 "sourceUrl": "https://github.com/scala/scala",
                 "homepage": "https://www.scala-lang.org/",
                 "currentVersion": "2.13.8",
                 "fixedVersion": "2.13.8"
               },
               {
                 "registryUrls": ["https://repo.maven.apache.org/maven2"],
                 "depName": "com.typesafe.scala-logging:scala-logging",
                 "packageName": "com.typesafe.scala-logging:scala-logging_2.13",
                 "currentValue": "3.9.4",
                 "datasource": "sbt-package",
                 "depIndex": 1,
                 "updates": [],
                 "warnings": [],
                 "versioning": "ivy",
                 "sourceUrl": "https://github.com/lightbend/scala-logging",
                 "dependencyUrl": "https://repo.maven.apache.org/maven2/com/typesafe/scala-logging",
                 "currentVersion": "3.9.4",
                 "fixedVersion": "3.9.4"
               },
               {
                 "registryUrls": ["https://repo.maven.apache.org/maven2"],
                 "depName": "ch.qos.logback:logback-classic",
                 "packageName": "ch.qos.logback:logback-classic",
                 "currentValue": "1.2.10",
                 "datasource": "sbt-package",
                 "depIndex": 2,
                 "updates": [
                   {
                     "bucket": "latest",
                     "newVersion": "1.2.11",
                     "newValue": "1.2.11",
                     "newMajor": 1,
                     "newMinor": 2,
                     "updateType": "patch",
                     "branchName": "renovate/all-versions"
                   }
                 ],
                 "warnings": [],
                 "versioning": "ivy",
                 "dependencyUrl": "https://repo.maven.apache.org/maven2/ch/qos/logback",
                 "currentVersion": "1.2.10",
                 "isSingleVersion": true,
                 "fixedVersion": "1.2.10"
               }
             ]
           }
         ]
       }
DEBUG: processRepo() (repository=nilius/renovate_issue)
DEBUG: Processing 1 branch: renovate/all-versions (repository=nilius/renovate_issue)
```
</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by hasanwhitesource**
## Changes

Before parsing we are now replacing '=\n' with '=' at the end of each line to make the parsing work in case of having a definition for a variable and then setting its value in a new line.

## Context

Closes #14846 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/sbt/extract.ts [hunk 1]
```diff
@@ -347,7 +347,9 @@ export function extractPackageFile(content: string): PackageFile | null {
   if (!content) {
     return null;
   }
-  const lines = content.split(regEx(/\n/)).map(stripComment);
+  const equalsToNewLineRe = regEx(/=\s*\n/, 'gm');
+  const goodContentForParsing = content.replace(equalsToNewLineRe, '=');
+  const lines = goodContentForParsing.split(regEx(/\n/)).map(stripComment);
```
Comment (Gabriel-Ladzaretti) [inline lib/modules/manager/sbt/extract.ts:352]:
```suggestion
  const goodContentForParsing = content.replace(equalsToNewLineRe, '=');
  const lines = goodContentForParsing.split(newlineRegex).map(stripComment);
```

please use `newlineRegex` from `util/regex` module

**Review by viceice — COMMENTED** [review]
see suggestions, otherwise LGTM

**Comment by hasanwhitesource** [comment]
@rarkins I think we are done here

File: lib/modules/manager/sbt/extract.ts [hunk 2]
```diff
@@ -347,7 +347,9 @@ export function extractPackageFile(content: string): PackageFile | null {
   if (!content) {
     return null;
   }
-  const lines = content.split(regEx(/\n/)).map(stripComment);
+  const equalsToNewLineRe = regEx(/=\s*\n/, 'gm');
```
Comment (zharinov) [inline lib/modules/manager/sbt/extract.ts:350]:
```suggestion
  const equalsToNewLineRe = regEx(`=\\s*${newlineRegex.source}`, 'gm');
```

**Review by zharinov — CHANGES_REQUESTED** [review]

**Review by zharinov — APPROVED** [review]
_(deleted my previous suggestion)_

LGTM

**Review by Gabriel-Ladzaretti — APPROVED** [review]
LGTM

**Review by rarkins — APPROVED** [review]

**Pull request merged**